### PR TITLE
Fixes #36 - Add beforeCreate and afterCreate hooks to arc-create.

### LINKED ISSUE #36: [Proposal] Add beforeCreate afterCreate hooks to arc-create
## Problem

Currently, arc-deploy supports `beforeDeploy` and `afterDeploy` hooks, which allow developers to create plugins. There may be plugins that need to be run before creation and before deploy. In my case, when using [arc-plugin-marko](https://github.com/austinkelleher/arc-plugin-marko), an app needs to run the plugin code before creation otherwise the first deploy (from the create step) will deploy broken code (non-compiled templates, no prebuilt pages, no assets uploaded to S3 etc).

## Proposal

Expose `beforeCreate` and `afterCreate` hooks.

I would be happy to work on this feature if we decide it's appropriate.

/cc @brianleroux

### PR DESCRIPTION
**Pull request by austinkelleher**
/cc @brianleroux

### CODE REVIEW

File: src/create/index.js [hunk 1]
```diff
@@ -28,20 +30,29 @@ module.exports = function generate(params, callback) {
     function _arcFileValid(arc, callback) {
       validate(arc, callback)
     },
+    function _beforeCreate(arc, callback) {
```
Comment (austinkelleher) [inline src/create/index.js]: I think we should revisit this whole file and clean it up to be more like the new deploy index, but I didn't think this should be in the same PR.

File: src/util/runPluginFunction.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
+/* eslint global-require:"off" */
+/**
+ * calls a specific plugin function in any plugins registered in .arc
+ */
+module.exports = function runPluginFunction(params, functionName) {
```
Comment (austinkelleher) [inline src/util/runPluginFunction.js]: Pulled out this code into a new file that can be shared between the deploy and create commands.

File: src/create/cli.js [hunk 3]
```diff
@@ -6,6 +6,10 @@ generate({
   arcFile: path.join(process.cwd(), '.arc'),
   execute: true
 },
-function() {
-  console.log('\n')
+function(err) {
+  if (err) {
+    console.error('Error creating resources: ', err)
```
Comment (austinkelleher) [inline src/create/cli.js]: Bonus change: Errors weren't getting logged out if the CLI failed for some reason.

File: src/create/after-create.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+/* eslint global-require:"off" */
+var runPluginFunction = require('../util/runPluginFunction')
+/**
+ * calls any plugins registered in .arc with afterCreate
+ */
+module.exports = function afterCreate(params, callback) {
```
Comment (austinkelleher) [inline src/create/after-create.js]: I'm not a huge fan of mixing callbacks and promises, so I made the `util/runPluginFunction` always return a promise and here we just wait for it to be complete to call the original callback.

**Comment by brianleroux** [comment]
Looks great @austinkelleher! Added you here and to npm as a member so (I think) you can merge/patch/publish now as you see fit! Thx for this work 👊 💕

**Pull request merged**